CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+  - `fix typo in stuff`
+  - `fix compilation`
+  - `change stuff a bit`
+  - and so on...
+
+##### `rebase` in practice
+
+`git log --graph --oneline` shows your commit history as graph.
+To make some changes in that graph, you do an **interactive rebase**:
+
+```
+git rebase -i -m upstream/master
+```
+
+With this command, your new "base" is `upstream/master` and you can
+then change any of your branch's commits.
+
+`-i` will open an interactive editor where you can choose actions for each individual commit:
+
+- re-order commits
+- drop commits by deleting their line
+- squash/fixup ("meld") your commits
+- reword a commit message
+- stop rebasing at a commit to edit (`--amend`) it manually
+
+Just follow the messages on screen.
+
+##### Changing commits with `amend` and `fixup`
+
+There's also `git commit --amend` which is a "mini-rebase" that modifies just the last commit with your current changes by `git add`.
+It just skips the creation of a new commit and instead melds the changes into the last one you made.
+
+If you want to update a single commit in the range `[upstream/master, current HEAD]` which is not the last commit:
+
+- `edit stuff you wanna change in some previous commit`
+- `git add changed_stuff`
+- `git commit --fixup $hash_of_commit_to_be_fixed`
+- `git rebase --autosquash -i -m upstream/master`
+
+##### Pushing changes
+
+After you have rebased stuff (["rewritten history"](https://www.youtube.com/watch?v=9lXuZHkOoH8)) that had already been pushed,
+git will not accept your pushes because they're not simple fast-forwards:
+
+- The commit contents and the parent commit have changed as you updated the commit, therefore the commit hash changed, too.
+  - If somebody used those commits, they will keep a copy
+    and have a hard time updating to your updated version (because they "use" the old hashes).
+  - Update your pull request branch with your re-written history!
+
+- **force push** is the standard way of overwriting your development work with the fixed and mergeable version of your contribution!
+  - Why? You changed the commits, so you want the old ones to be deleted!
+
+  You can use any of:
+  - `git push origin +my-awesome-feature`
+  - `git push origin -f my-awesome-feature`
+  - `git push origin --force my-awesome-feature`
+
+Some extra tutorials on `git rebase`:
+
+- [Atlassian's Git Tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/)
+- [Pro Git book](http://git-scm.com/book)
+- `man git-rebase`
+
+<!-- TODO?
+## Development Setup
+
+If you want to set up a local development environment, follow the steps outlined in the [README.md](README.md) file.
+-->
+
+## License
+
+By contributing to `rustic` or any crates contained in this repository, you agree that your contributions will be licensed under either of:
+
+- [Apache License, Version 2.0](./LICENSE-APACHE)
+- [MIT license](./LICENSE-MIT)
+
+at your option. Unless you explicitly state otherwise, any contribution intentionally submitted
```

Comment:
I think that should be something like "at the option of everyone using `rustic`.

Reply:
Oh, I somehow missed your question, sorry for that!
We want to have rustic dual-licenced which means that the *user* of rustic may choose which licence to use. That means, that we must require contriubtors to also have their work dual-licenced in the same sense. If we write "you agree that your contributions will be licensed under either of:

- [Apache License, Version 2.0](./LICENSE-APACHE)
- [MIT license](./LICENSE-MIT)

at your option." I think this can be misunderstanding that a contributor may choose the licence under which their work will be licenced. But this is not what we want.